File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/cert/SwlCertExchanger.java
package i2f.swl.cert;

import i2f.swl.cert.data.SwlCert;
import i2f.swl.cert.data.SwlCertPair;
import i2f.swl.cert.impl.SwlResourceCertManager;
import i2f.swl.core.exchanger.SwlExchanger;
import i2f.swl.data.SwlData;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * @author Ice2Faith
 * @date 2024/8/8 16:32
 * @desc
 */
@Data
@NoArgsConstructor
public class SwlCertExchanger extends SwlExchanger {
    protected SwlCertManager certManager = new SwlResourceCertManager();

    public void createCertPair(String certId) {
        SwlCertPair pair = generateCertPair(certId);
        certManager.storePair(pair);
    }

    public SwlData sendByCertId(String certId,
                                List<String> parts) {
        SwlCert cert = certManager.loadClient(certId);
        return sendByCert(cert, parts);
    }

    public SwlData sendByCertId(String certId,
                                List<String> parts,
                                List<String> attaches) {
        SwlCert cert = certManager.loadClient(certId);
        return sendByCert(cert, parts, attaches);
    }

    public SwlData receiveByCertId(SwlData request,
                                   String certId) {
        SwlCert cert = certManager.loadClient(certId);
        return receiveByCert(request, cert);
    }

    public SwlData receiveByCertId(SwlData request,
                                   String certId,
                                   String clientId) {
        SwlCert cert = certManager.loadClient(certId);
        return receiveByCert( request, cert,clientId);
    }


    public SwlData responseByCertId(String certId,
                                    List<String> parts) {
        SwlCert cert = certManager.loadServer(certId);
        return sendByCert(cert, parts);
    }

    public SwlData responseByCertId(String certId,
                                    List<String> parts,
                                    List<String> attaches) {
        SwlCert cert = certManager.loadServer(certId);
        return sendByCert(cert, parts, attaches);
    }

    public SwlData acceptByCertId(SwlData request,
                                  String certId) {
        SwlCert cert = certManager.loadServer(certId);
        return receiveByCert(request, cert);
    }

    public SwlData acceptByCertId(SwlData request,
                                  String certId,
                                  String clientId) {
        SwlCert cert = certManager.loadServer(certId);
        return receiveByCert( request, cert,clientId);
    }


}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/core/cert/SwlCertExchanger.js
import SwlExchanger from "../core/exchanger/SwlExchanger";
import SwlStorageCertManager from "./impl/SwlStorageCertManager";

/**
 * @param certManager {SwlCertManager}
 * @return {SwlCertExchanger}
 * @extends {SwlExchanger}
 * @constructor {SwlCertExchanger}
 */
function SwlCertExchanger(certManager=new SwlStorageCertManager()) {
    /**
     *
     * @type {SwlCertManager}
     */
    this.certManager=certManager
}


// 继承
SwlCertExchanger.prototype = Object.create(SwlExchanger.prototype)
SwlCertExchanger.prototype.constructor = SwlCertExchanger


/**
 * @param certId {String}
 * @return {void}
 */
SwlCertExchanger.prototype.createCertPair = function (certId) {
    let pair = this.generateCertPair(certId);
    this.certManager.storePair(pair);
}
/**
 * @param certId {String}
 * @param parts {string[]|null}
 * @param attaches {string[]|null}
 * @return {SwlData}
 */
SwlCertExchanger.prototype.sendByCertId = function (certId,parts,attaches=[]) {
    let cert = this.certManager.loadClient(certId);
    return this.sendByCert(cert, parts, attaches);
}
/**
 * @param request {SwlData}
 * @param certId {String}
 * @param clientId {String|null}
 * @return {SwlData}
 */
SwlCertExchanger.prototype.receiveByCertId = function (request,certId,clientId=null) {
    let cert = this.certManager.loadClient(certId);
    if(!clientId){
        clientId=cert.certId
    }
    return this.receiveByCert(request,cert,clientId);
}

/**
 * @param certId {String}
 * @param parts {string[]|null}
 * @param attaches {String[]|null}
 * @return {SwlData}
 */
SwlCertExchanger.prototype.responseByCertId = function (certId,parts,attaches=[]) {
    let cert = this.certManager.loadServer(certId);
    return this.sendByCert(cert, parts, attaches);
}

/**
 * @param request {SwlData}
 * @param certId {String}
 * @param clientId {String|null}
 * @return {SwlData}
 */
SwlCertExchanger.prototype.acceptByCertId = function (request,certId,clientId=[]) {
    let cert = this.certManager.loadServer(certId);
    if(!clientId){
        clientId=certId
    }
    return this.receiveByCert(request, cert,clientId);
}

export default SwlCertExchanger
